Features chapter2

IIKUYY opened the pull request:
## what

- Agregar resumen del capitulo 2 junto con ejemplos de uso

## why

- Continuar con las tareas dadas

## references
- http://diveintohtml5.info/index.html

IIKUYY committed:
Feat: Actualizar TOC de navegación

## what

- Agregar vínculos a los menús

## why

- Mantener el uso correcto de TOC de navegación

## references
-

IIKUYY committed:
Docs: Agregar resumen del capitulo 2

## what

- Agregar el resumen del capitulo 2 junto con sus ejemplos

## why

- Continuar con las tareas dadas

## references
- http://diveintohtml5.info/index.html

IIKUYY commented:
Para hacer los ejemplos use una mezcla de búsquedas de google, IA y mis conocimientos, esto debido a que aun no manejo del todo bien JavaScript, si encuentra algo que se pueda optimizar o este mal hecho por favor hágamelo saber

semanticmx commented on Ch2/Ch2.md:
````diff
@@ -0,0 +1,118 @@
+```
+<!DOCTYPE html>
+<html>
+<head>
````
como en este curso usamos un poco más de código voy a ir agregando correcciones para que las apliques en el PR actual y en los siguientes, estas correcciones van formando nuestra guía de estilo conforme las vamos haciendo.

https://github.com/semanticmx/html5-coding-standards

La puedes ir revisando y aplicando conforme avances con el curso

El primer cambio que necesitamos aquí es identar los elementos dentro del tag html.

actual
```
<html>
<head>
...
</body>
<
```

deseado
```
<html>
  <head>
  ...
  </body>
<
```

semanticmx commented on Ch2/Metodos/Metodo1/M1.html:
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE <!DOCTYPE html>
+<html lang="en">
+    <head>
```
acá la identación quedo perfecto, los IDEs te ayudan con eso.

semanticmx commented on Ch2/Metodos/Metodo1/M1.js:
```diff
@@ -0,0 +1,11 @@
+function supports_geolocation() {
```
Si tu usas la tecla de TAB en el teclado para identar, los espacios siempre van a ser los mismos y si configuras el IDE para que siempre te convierta los TAB's a 2 o 4 espacios entonces no habrá problemas de que las líneas de código queden identadas a diferentes espacios.

semanticmx commented on Ch2/Metodos/Metodo2/M2.js:
```diff
@@ -0,0 +1,18 @@
+      document.write("El navegador no soporta texto en canvas.");
+    }
+  });
+function supports_canvas_text() {
```
para los métodos en JavaScript vamos a usar camelCase

en lugar de supportsCanvasText

que diga: supportsCanvasText

semanticmx commented on Ch2/Metodos/Metodo2/M2.js:
```diff
@@ -0,0 +1,18 @@
+  });
+function supports_canvas_text() {
+    if (!supports_canvas()) { return false; }
+    var dummy_canvas = document.createElement('canvas');
```
lo mismo para nombres de variables

IIKUYY committed:
Chores: realizar cambios necesarios

## what

- Cambiar la semantica de Java a camelCase
- Corregir la identación

## why

- Arreglar la semantica de los archivos

## references
-
